src/clients/ProfitClient.ts

```diff
@@ -121,7 +127,13 @@ export class ProfitClient {
     try {
       cgPrices = await this.coingeckoPrices(Object.keys(l1Tokens));
     } catch (err) {
-      this.logger.warn({ at: "ProfitClient", message: "Failed to retrieve prices.", err, l1Tokens });
+      const addresses: Array<string> = Object.keys(l1Tokens);
+      this.logger.warn({
+        at: "ProfitClient",
+        message: `Failed to retrieve token prices (${err}).`,
```

Comment:
I would just make `err` a key in the log like so
```suggestion
        message: `Failed to retrieve token prices.`,
        err,
```

Reply:
This is what I had originally, but the contents of err never gets included in the log message - it only prints "err". I suspect that's because it's a string and winston just skips it - does that sound plausible?

Reply:
> This is what I had originally, but the contents of err never gets included in the log message - it only prints "err". I suspect that's because it's a string and winston just skips it - does that sound plausible?

So it's not "just" a string..it seems a bit maleable, depending on where it's used. Including it in the log directly as "error: err" blows up and dumps the corresponding stack trace _after_ the end of the log:

```
2022-08-17 19:52:19 [warn]: {                                                                                                                                                                             [285/1799]
  "at": "ProfitClient",                                                                                                                                                                                             
  "message": "Failed to retrieve token prices.",                                                                                                                                                                    
  "tokens": [                                                                                                                                                                                                       
    {                                                                                                                                                                                                               
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",                                                                                                                                                      
      "symbol": "WETH",                                                                                                                                                                                             
      "decimals": 18                                                                                                                                                                                                
    },                                                                                                                                                                                                              
    {                                                                                                                                                                                                               
      "address": "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",                                                                                                                                                      
      "symbol": "USDC",                                                                                                                                                                                             
      "decimals": 6                                                                                                                                                                                                 
    },                                                                                                                                                                                                              
    {                                                                                                                                                                                                               
      "address": "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",                                            
      "symbol": "WBTC",                       
      "decimals": 8                                                                         
    },                                        
    {                                                                                       
      "address": "0x04Fa0d235C4abf4BcF4787aF4CF447DE572eF828",                                            
      "symbol": "UMA",                        
      "decimals": 18                                                                        
    },                                                                                                    
    {                                                                                       
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",                                            
      "symbol": "DAI",                                                                                    
      "decimals": 18                          
    },                                                                                      
    {                                                                                       
      "address": "0x42bBFa2e77757C645eeaAd1655E0911a7553Efbc",                                            
      "symbol": "BOBA",                                                                     
      "decimals": 18                          
    },                                                                                      
    {                                                                                       
      "address": "0x3472A5A71965499acd81997a54BBA8D852C6E53d",                              
      "symbol": "BADGER",                                                                   
      "decimals": 18                                                                                      
    },                                        
    {                                                                                       
      "address": "0xba100000625a3754423978a60c9317c58a424e3D",                                            
      "symbol": "BAL",                                                                      
      "decimals": 18                                                                                      
    }                                                                                       
  ],                                          
  "bot-identifier": "XXXX"                                                              
}                                                                                                         
Error: Unknown Coingecko Error                
    at Coingecko.call (/home/pxrl/dev/across-protocol/relayer-v2/node_modules/@uma/sdk/dist/src/coingecko/coingecko.ts:97:13)                                                                                       
    at runMicrotasks (<anonymous>)                                                          
    at processTicksAndRejections (node:internal/process/task_queues:96:5)                                 
    at async Coingecko.getContractPrices (/home/pxrl/dev/across-protocol/relayer-v2/node_modules/@uma/sdk/dist/src/coingecko/coingecko.ts:75:28)                                                                    
    at async ProfitClient.coingeckoPrices (/home/pxrl/dev/across-protocol/relayer-v2/src/clients/ProfitClient.ts:174:12)                                                                                            
    at async ProfitClient.update (/home/pxrl/dev/across-protocol/relayer-v2/src/clients/ProfitClient.ts:134:18)                                                                                                     
    at async updateClients (/home/pxrl/dev/across-protocol/relayer-v2/src/common/ClientHelper.ts:176:3)   
    at async updateRelayerClients (/home/pxrl/dev/across-protocol/relayer-v2/src/relayer/RelayerClientHelper.ts:64:3)                                                                                               
    at async Object.runRelayer [as relayer] (/home/pxrl/dev/across-protocol/relayer-v2/src/relayer/index.ts:29:7)                                                                                                   
    at async run **(/home/pxrl/dev/across-protocol/relayer-v2/index.ts:38:9)**
```

Reply:
@pxrl That looks more like some print that's happening _outside of_ your log since it's outside of the json.

There's a lot of formatting that's happening [behind the scenes](https://github.com/UMAprotocol/protocol/tree/master/packages/financial-templates-lib/src/logger) with the logger. One tip that may magically solve this is if you rename err in your object to error (there's some special handling in the logger for the field error).

Basically just this:
```ts
this.logger.warn({
...
  error: err,
...
});
```

Reply:
Resolved in DM - this is a known quirk in the logging implementation. Will proceed with embedding err directly in the message - this behaves fine in my testing.

https://github.com/UMAprotocol/protocol/blob/8c47a3bccb302d46ad22ee31ccc93090dcc79164/packages/financial-templates-lib/src/logger/ConsoleTransport.ts#L20-L25

```
2022-08-19 20:19:14 [warn]: {                        
  "at": "ProfitClient",                              
  "message": "Failed to retrieve token prices (Error: Unknown Coingecko Error).",                         
  "tokens": [                                        
    "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",                                                         
    "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48",                                                         
    "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",                                                         
    "0x04Fa0d235C4abf4BcF4787aF4CF447DE572eF828",                                                         
    "0x6B175474E89094C44Da98b954EedeAC495271d0F",                                                         
    "0x42bBFa2e77757C645eeaAd1655E0911a7553Efbc",                                                         
    "0x3472A5A71965499acd81997a54BBA8D852C6E53d",                                                         
    "0xba100000625a3754423978a60c9317c58a424e3D"                                                          
  ],                                                 
  "bot-identifier": "XXXX"                       
}
```